### PR TITLE
feat: Healthz api check appId support

### PR DESCRIPTION
**Pull request by alberthuang24**
related: https://github.com/dapr/components-contrib/issues/2489

Signed-off-by: AlbertHuang <r.albert.huang@gmail.com>

# Description

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by ItalyPaleAle** [comment]
CC: @DeepanshuA since you've been looking into the healthz API

**Comment by DeepanshuA** [comment]
@alberthuang24 I am not able to understand the use-case behind this work.
When I was working on components Health check API work, then it was discussed and realized that we may want rather different query params in future with existing healthz check api itself.
Can you please raise an issue and elaborate that how does having {appid} in url helps, and for what use-cases. I think there-in it may be concluded the best approach forward, and then a PR can follow.

**Comment by alberthuang24** [comment]
Hello @DeepanshuA 

This support was added to allow external parties to pass the expected `app-id`. 
This method allows external applications to determine if some old IP address is still the expected `app-id`

This prevents the `consul name-resolution component` from incorrectly resolving the service to another `app-id` IP address as mentioned in https://github.com/dapr/components-contrib/issues/2489

**Comment by alberthuang24** [comment]
Maybe we could also add a separate entry for the checksum. Perhaps called `healthz/expect/{app-id}`

**Comment by ItalyPaleAle** [comment]
hi @alberthuang24 would the solution implemented in #5914, which will ship with Dapr 1.11, work for your scenario?

**Comment by alberthuang24** [comment]
> hi @alberthuang24 would the solution implemented in #5914, which will ship with Dapr 1.11, work for your scenario?

This #5914 only addresses local issues, I need to use `consul-dns` on kubernetes

**Comment by ItalyPaleAle** [comment]
As discussed in https://github.com/dapr/components-contrib/pull/2490, the intent of this PR can be better addressed in a different way, as per https://github.com/dapr/components-contrib/pull/2594

**Comment by alberthuang24** [comment]
@ItalyPaleAle Is There an updated conclusion?

File: pkg/http/api.go [hunk 1]
```diff
@@ -475,7 +475,7 @@ func (a *api) constructHealthzEndpoints() []Endpoint {
 	return []Endpoint{
 		{
 			Methods:       []string{fasthttp.MethodGet},
-			Route:         "healthz",
+			Route:         "healthz/{appId:*}",
```
Comment (artursouza) [inline pkg/http/api.go:478]: Can we use query param instead?

**Review by artursouza — CHANGES_REQUESTED** [review]

**Comment by ItalyPaleAle** [comment]
> Is There an updated conclusion?

@alberthuang24 We spoke about this during the maintainers call this morning. I'm ok with this approach and so is @artursouza .

However, Artur asked (see his review) if we can pass this as query string argument rather than URL path, since it can allow more flexibility in the future.

**Comment by alberthuang24** [comment]
> > Is There an updated conclusion?
> 
> @alberthuang24 We spoke about this during the maintainers call this morning. I'm ok with this approach and so is @artursouza .
> 
> However, Artur asked (see his review) if we can pass this as query string argument rather than URL path, since it can allow more flexibility in the future.

done.

**Comment by ItalyPaleAle** [comment]
@alberthuang24

I wasn't able to make changes into your branch (you didn't turn on "allow maintainers to make changes"), so please do this:

```sh
git pull https://github.com/italyPaleAle/dapr update-5868
```

Among other things, this changes the query string parameter to `appid` (we don't normally put dashes in it, such as `app-id`).

Then I'll approve

**Comment by alberthuang24** [comment]
> @alberthuang24
> 
> I wasn't able to make changes into your branch (you didn't turn on "allow maintainers to make changes"), so please do this:
> 
> ```shell
> git pull https://github.com/italyPaleAle/dapr update-5868
> ```
> 
> Among other things, this changes the query string parameter to `appid` (we don't normally put dashes in it, such as `app-id`).
> 
> Then I'll approve

done

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by ItalyPaleAle — APPROVED** [review]

**Review by daixiang0 — COMMENTED** [review]
Please update the base, thanks.

**Comment by alberthuang24** [comment]
done.

**Comment by ItalyPaleAle** [comment]
@alberthuang24 please enable this checkmark:

image

This way we can ensure the branch is up-to-date and merge it

**Comment by ItalyPaleAle** [comment]
It would be here, under the participants:

image

Maybe it's because you've opened this PR from your master branch?

We could either:

- Close this PR and re-open it from a different branch
- Re-open this PR for you (giving you credit in the commits)

**Comment by mukundansundar** [comment]
> It would be here, under the participants:
> 
> image
> 
> Maybe it's because you've opened this PR from your master branch?
> 
> We could either:
> 
> * Close this PR and re-open it from a different branch
> * Re-open this PR for you (giving you credit in the commits)

@alberthuang24 Can you reopen this PR after updating the latest changes from master? 
Please do give permissions to maintainers to update the PR

**Pull request closed without merging**